### PR TITLE
Add template to create Azure SQL DB and import bacpac

### PR DESCRIPTION
**Pull request by yochananrachamim**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

Adding Template to create Azure SQL Server and Database while we set the backup storage (Local, Zone or global) and importing a bacpac file. 
this is needed as the import options that we have does not allows us to set the backup storage to be used, in case or a policy that enforce local backup storage redundancy every import fail as it tries to create database with the default GRS backup storage.
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

File: 101-sql-database-Import/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          "location": "[parameters('location')]",
+          "properties":
+          {
+            "storageAccountType": "[parameters('storageAccountType')]"
+          },
+          "sku": {
+            "name": "Standard",
+            "tier": "Standard"
+          },
+          "dependsOn": [
```
Comment (bmoore-msft) [inline 101-sql-database-Import/azuredeploy.json:96]: move this above properties

File: 101-sql-database-Import/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            "[resourceId('Microsoft.Sql/servers', concat(parameters('serverName')))]"
+          ],
+          "resources": [
+                {
+                    "condition":"[not(empty(parameters('bacpacUrl')))]",
```
Comment (bmoore-msft) [inline 101-sql-database-Import/azuredeploy.json:101]: this needs to be tested, you can check other samples in the repo for a bacpac you can use

File: 101-sql-database-Import/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "serverName": {
+      "value": "GEN-UNIQUE"
+    },
+    "administratorLogin": {
+      "value": "GEN-UNIQUE"
+    },
+    "administratorLoginPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "storageAccountType": {
```
Comment (bmoore-msft) [inline 101-sql-database-Import/azuredeploy.parameters.json:14]: remove the params that have defaultValues in the template

File: 101-sql-database-Import/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Azure SQL Database
+
+![Azure Public Test Date](https://azurequickstartsservice.blob.core.windows.net/badges/101-sql-database-import/PublicLastTestDate.svg)
```
Comment (bmoore-msft) [inline 101-sql-database-Import/README.md:3]: check the pipeline results for changes that need to be made here (casing issue in the url)

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-sql-database-Import/azuredeploy.json (hunk 2 above)
Comment (yochananrachamim) [reply 101-sql-database-Import/azuredeploy.json:101]: it was tested and is successful, this allows users to import to a database while we set the backup storage redundancy. 
this is not implemented yet for any of the import options (Portal, SQLPackage) this is why this is so important

File: 101-sql-database-Import/azuredeploy.json (hunk 1 above)
Comment (yochananrachamim) [reply 101-sql-database-Import/azuredeploy.json:96]: Done

File: 101-sql-database-Import/azuredeploy.parameters.json (hunk 3 above)
Comment (yochananrachamim) [reply 101-sql-database-Import/azuredeploy.parameters.json:14]: Done

File: 101-sql-database-Import/README.md (hunk 4 above)
Comment (yochananrachamim) [reply 101-sql-database-Import/README.md]: Done

File: 101-sql-database-Import/azuredeploy.parameters.json (hunk 3 above)
Comment (yochananrachamim) [reply 101-sql-database-Import/azuredeploy.parameters.json:14]: Done

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

**Comment by bmoore-msft** [comment]
@yochananrachamim - check this PR for updates that may be needed to documentation that references this sample.

File: 101-sql-database-Import/azuredeploy.json (hunk 2 above)
Comment (bmoore-msft) [reply 101-sql-database-Import/azuredeploy.json:100]: Not sure I follow re: NYI... if it's not implemented in the platform we should remove it from the sample.  If it is in the platform we want to test it in the pipeline.

Comment (yochananrachamim) [reply 101-sql-database-Import/azuredeploy.json:100]: This template should be used by customers that need to import Azure SQL DB where policy enforced to use local redundant storage for backups. 
the regular import interface (using portal or command line tools) are not yet implementing this option. 
this is why this option is so important. 
here is a blog post I published about it: https://techcommunity.microsoft.com/t5/azure-database-support-blog/import-database-when-policy-quot-azure-sql-db-should-avoid-using/ba-p/1960857

Comment (bmoore-msft) [reply 101-sql-database-Import/azuredeploy.json:100]: Thanks, that helps clarify... We're providing a sample for the import case, we just need to be able to test that code.  Here are the changes we need:

1) the import resource should not be conditional - if it is, this is just a template that creates a db and a server and there are plenty of samples that do that already
2) the import resource needs to be tested, there are a number of samples that do this already, you can look at those for how it's done...  Essentially you can use a sasUri for the bacpac and provide it with the sample.  You can provide the option to use a StorageKey, but CI will use the sas key.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**